code/game/objects/items/storage/toolbox.dm

```diff
@@ -20,22 +20,59 @@
 	material_flags = MATERIAL_EFFECTS | MATERIAL_COLOR
 	var/latches = "single_latch"
 	var/has_latches = TRUE
+	var/latch_exempt = FALSE
 	wound_bonus = 5
 
 /obj/item/storage/toolbox/Initialize()
 	. = ..()
+	if(prob(5))
+		has_latches = FALSE
 	if(has_latches)
 		if(prob(10))
 			latches = "double_latch"
 			if(prob(1))
 				latches = "triple_latch"
 	update_appearance()
 
+/obj/item/storage/toolbox/examine(mob/user)
+	. = ..()
+	if(!has_latches && !latch_exempt)
+		. += span_danger("\The [src] lacks a latch, be careful to not spill its contents.")
+
 /obj/item/storage/toolbox/update_overlays()
 	. = ..()
 	if(has_latches)
 		. += latches
 
+/obj/item/storage/toolbox/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	. = ..()
+	if(!has_latches && !latch_exempt)
+		var/list/obj/item/oldContents = contents.Copy()
+		SEND_SIGNAL(src, COMSIG_TRY_STORAGE_QUICK_EMPTY)
+		for(var/obj/item/scatter_item in oldContents)
+			INVOKE_ASYNC(src, .proc/do_scatter, scatter_item)
+
+/obj/item/storage/toolbox/dropped(mob/user)
+	. = ..()
+	if(!has_latches && !latch_exempt)
+		var/list/obj/item/oldContents = contents.Copy()
+		SEND_SIGNAL(src, COMSIG_TRY_STORAGE_QUICK_EMPTY)
+		for(var/obj/item/scatter_item in oldContents)
+			INVOKE_ASYNC(src, .proc/do_scatter, scatter_item)
+
+/obj/item/storage/toolbox/attack(mob/living/M, mob/living/user)
+	. = ..()
+	if(!has_latches && !latch_exempt)
+		var/list/obj/item/oldContents = contents.Copy()
+		SEND_SIGNAL(src, COMSIG_TRY_STORAGE_QUICK_EMPTY)
+		for(var/obj/item/scatter_item in oldContents)
+			INVOKE_ASYNC(src, .proc/do_scatter, scatter_item)
+
+/obj/item/storage/toolbox/proc/do_scatter(obj/item/scatter_item)
+	for(var/i in 1 to rand(1,2))
+		if(scatter_item)
+			step(scatter_item, pick(NORTH,SOUTH,EAST,WEST))
+			sleep(1)
```

Comment:
Change to `stoplag(1)`, it's what you should be using instead of `sleep(1)`

Reply:
I copied this code direct from food tray scatter code. just FYI, if you want someone to change that over